cmd/nerdctl/run.go

```diff
@@ -635,13 +638,31 @@ func createContainer(cmd *cobra.Command, ctx context.Context, client *containerd
 	var s specs.Spec
 	spec := containerd.WithSpec(&s, opts...)
 	cOpts = append(cOpts, spec)
-
 	logrus.Debugf("final cOpts is %v", cOpts)
 	container, err := client.NewContainer(ctx, id, cOpts...)
 	if err != nil {
-		return nil, "", nil, err
+		gcContainer := func() {
+			var isErr bool
```

Comment:
Can this be an error type; and then we can wrap errors and then check
```
if err != nil {
  logrus.Warnf("failed to remove container because of %s", err.Error())
}.
```
to provide better info to debug?